flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriterStateSerializerTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.sink.writer;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** Test class for {@link AsyncSinkWriterStateSerializer}. */
+public class AsyncSinkWriterStateSerializerTest {
+
+    @Test
+    public void testSerializeAndDeSerialize() throws IOException {
+        AsyncSinkWriterStateSerializerImpl stateSerializer =
+                new AsyncSinkWriterStateSerializerImpl();
+        List<String> testData = getTestData(100);
+        BufferedRequestState<String> state = wrapStringRequests(testData);
+        Collection<String> deserializedState =
+                stripStringEntryWrappers(
+                        stateSerializer.deserialize(0, stateSerializer.serialize(state)));
+
+        Assertions.assertThat(deserializedState).isEqualTo(testData);
+    }
+
+    private List<String> getTestData(int size) {
+        List<String> testData = new ArrayList<>();
+        for (int i = 1; i <= size; i++) {
+            testData.add(String.format("value:%d", i));
+        }
+        return testData;
+    }
```

Comment:
This is duplicated in a few places. Can we move to a test util? Also can simplify with java streams as pointed out in a nit comment above